api/src/pcapi/core/users/external/user_automations.py

```diff
@@ -0,0 +1,166 @@
+from datetime import date
+from datetime import datetime
+from typing import List
+
+from dateutil.relativedelta import relativedelta
+from sqlalchemy import func
+from sqlalchemy.orm import Query
+from sqlalchemy.orm import load_only
+from sqlalchemy.sql.expression import and_
+from sqlalchemy.sql.expression import or_
+
+from pcapi import settings
+from pcapi.core.payments.models import Deposit
+from pcapi.core.users.constants import ELIGIBILITY_AGE_18
+from pcapi.core.users.external.sendinblue import add_contacts_to_list
+from pcapi.models import User
+
+
+YIELD_COUNT_PER_DB_QUERY = 1000
+
+
+def get_young_users_emails_query() -> Query:
+    # Not clear who a young user is: a beneficiary or any role not in (admin, pro, jouve)?
+    # .filter(User.is_beneficiary is True) would exclude users who did not complete registration process
+    return (
+        User.query.yield_per(YIELD_COUNT_PER_DB_QUERY)
+        .options(load_only(User.email))
+        .filter(User.has_pro_role.is_(False))
+        .filter(User.isAdmin.is_(False))
```

Comment:
(détail / style)

```suggestion
        .filter(User.has_admin_role.is_(False))
```

... devrait fonctionner, je pense ? C'est homogène avec ton utilisation de `has_pro_role` et ça prend de l'avance sur la future suppression de `isAdmin`. :)